### PR TITLE
Modernize tests under converters and visualization

### PR DESCRIPTION
**Pull request by rzhao271**
Affects #3743

Changes made in this Pull Request:
 - Changed `assert_almost_equal` to `assert_allclose` or `approx`

This PR only affects tests under the `converters` and `visualization` directories for now.

PR Checklist
------------
 - [x] Tests?
 - [x] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

### CODE REVIEW

File: package/CHANGELOG
Comment (orbeckst) [inline package/CHANGELOG]: I am not 100% sure if we need to include internal changes here as this is not particularly important for users — @IAlibay ??

In any case, add the issue number and PR as reference, please

File: testsuite/MDAnalysisTests/converters/test_openmm.py
Comment (orbeckst) [inline testsuite/MDAnalysisTests/converters/test_openmm.py:80]: That's a good equivalent test. For coordinates it makes sense to only use atol.

File: testsuite/MDAnalysisTests/converters/test_openmm.py
Comment (orbeckst) [inline testsuite/MDAnalysisTests/converters/test_openmm.py:96]: This is still used elsewhere. There are now tests failing because this is missing. Please check.

**Review by orbeckst — CHANGES_REQUESTED** [review]
Thank you for tackling this issue.

I only had time for an initial review. The primary problem is that there are now failing tests (probably because of the removal of `.prec`). Please address this problem and then ping me to review again. Thank you!

File: testsuite/MDAnalysisTests/converters/test_openmm.py
Comment (rzhao271) [reply testsuite/MDAnalysisTests/converters/test_openmm.py:96]: Sorry, I noticed but didn't think of reverting the PR back to a draft. Will check in a few hours!

File: package/CHANGELOG
Comment (IAlibay) [reply package/CHANGELOG]: Agreed, I would advocate for not including this in entry the changelog (sorry about that), please do keep your user name under the changelog authors though!

File: testsuite/MDAnalysisTests/converters/test_parmed.py
Comment (IAlibay) [inline testsuite/MDAnalysisTests/converters/test_parmed.py]: I'm not super fond of the formatting given we end up using a whole line for a two character variable. I won't block on that but I would ask to use f-strings and implicit continuation here instead of `+`

**Review by IAlibay — CHANGES_REQUESTED** [review]
just the one thing extra from what @orbeckst already mentioned

File: testsuite/MDAnalysisTests/converters/test_openmm.py
Comment (IAlibay) [inline testsuite/MDAnalysisTests/converters/test_openmm.py]: Can you use implicit continuation here? i.e.

```suggestion
            err_msg=("OpenMMBasicSimulationReader failed to get unitcell "
                     "dimensions from OpenMM Simulation Object"),
```

File: testsuite/MDAnalysisTests/converters/test_openmm.py
Comment (IAlibay) [inline testsuite/MDAnalysisTests/converters/test_openmm.py]: As above re; implicit line continuation instead of `+`

File: testsuite/MDAnalysisTests/converters/test_openmm.py
Comment (IAlibay) [inline testsuite/MDAnalysisTests/converters/test_openmm.py]: As above

File: testsuite/MDAnalysisTests/converters/test_parmed.py
Comment (IAlibay) [inline testsuite/MDAnalysisTests/converters/test_parmed.py]: As above

**Review by IAlibay — CHANGES_REQUESTED** [review]
just a couple of pep8 things - essentially the preference is to use implicit line continuation where possible. You could also just declare the error message before the assert_allclose call so as to make for a cleaner error message definition

**Review by orbeckst — APPROVED** [review]
Thank you for addressing all my comments. I'll leave it to @IAlibay to put the final approval on the PR and merge. Thank you very much for your first contribution!!

**Comment by orbeckst** [comment]
@IAlibay I assigned the PR to you, just so that it's clear that you're responsible for seeing it through.

**Review by IAlibay — APPROVED** [review]
awesome, thanks for your contribution @rzhao271 !

**Pull request merged**
